👊🏻Add gcsim commands

LeiShi1313 opened the pull request:
## 该 PR 相关 Issue / Involved issue

Close #707 


## 更改检查表 / Change Checklist
  
- [x] 有插件命令更新
  - [x] 已更新 bot 帮助文件
- [x] 有流程交互
  - [x] 指引明确
  - [ ] 可以退出
- [x] 会触发频率限制
  - [x] 有对应的限制措施
- [ ] 需要用户输入数据
  - [ ] 验证用户数据
  - [ ] 如果是文件，检查了文件大小
  - [ ] 对保存的数据再次进行了验证
- [x] 添加了新的依赖包
- [x] 测试
  - [x] 本地通过了测试
  - [ ] CI 通过了测试

## 说明 / Note

Commit:
:bug: Fix `user_command`

Commit:
:bug: Fix `group_command`

LeiShi1313 committed:
add genshin models and converters

LeiShi1313 committed:
add genshin models and converters

LeiShi1313 committed:
add genshin models and converters

Commit:
:memo: Add para notes

LeiShi1313 committed:
Add gcsim model and modified base model

LeiShi1313 committed:
Add gcsim model and modified base model

LeiShi1313 committed:
Add gcsim model and modified base model

LeiShi1313 committed:
write gcsim script parser

LeiShi1313 committed:
write gcsim script parser

LeiShi1313 committed:
write gcsim script parser

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in 857f23b according to the output
from Black.

Details: https://github.com/PaiGramTeam/PaiGram/pull/709

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in 857f23b according to the output
from Black.

Details: https://github.com/PaiGramTeam/PaiGram/pull/709

luoshuijs committed:
:bug: Fix `PYL-E0604`

luoshuijs committed:
:bug: Fix `PYL-E0604`

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in plugins/genshin/model/__init__.py with DeepSource Autofix

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in plugins/genshin/model/__init__.py with DeepSource Autofix

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in plugins/genshin/model/converters/enka.py with DeepSource Autofix

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in plugins/genshin/model/converters/enka.py with DeepSource Autofix

LeiShi1313 committed:
implement gcsim plugin

LeiShi1313 committed:
implement gcsim plugin

LeiShi1313 committed:
implement gcsim plugin

LeiShi1313 committed:
implement gcsim plugin

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in d928d8c according to the output
from Black.

Details: https://github.com/PaiGramTeam/PaiGram/pull/709

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in d928d8c according to the output
from Black.

Details: https://github.com/PaiGramTeam/PaiGram/pull/709

deepsource-autofix[bot] committed:
refactor: autofix issues in 3 files

Resolved issues in the following files with DeepSource Autofix:
1. modules/gcsim/file.py
2. plugins/genshin/gcsim/plugin.py
3. plugins/genshin/model/converters/enka.py

deepsource-autofix[bot] committed:
refactor: autofix issues in 3 files

Resolved issues in the following files with DeepSource Autofix:
1. modules/gcsim/file.py
2. plugins/genshin/gcsim/plugin.py
3. plugins/genshin/model/converters/enka.py

LeiShi1313 committed:
seperate gcsim runner

LeiShi1313 committed:
seperate gcsim runner

LeiShi1313 committed:
gcsim MVP done!

LeiShi1313 committed:
gcsim MVP done!

LeiShi1313 committed:
Merge branch 'gcsim' of github.com:PaiGramTeam/PaiGram into gcsim

LeiShi1313 committed:
remove gcdatabase first

LeiShi1313 committed:
remove gcdatabase first

Signed-off-by: Lei Shi <me@leishi.io>

deepsource-autofix[bot] committed:
refactor: autofix issues in 4 files

Resolved issues in the following files with DeepSource Autofix:
1. plugins/genshin/gcsim/plugin.py
2. plugins/genshin/gcsim/runner.py
3. plugins/genshin/model/base.py
4. plugins/genshin/model/converters/gcsim.py

luoshuijs commented on plugins/genshin/model/metadata.py:
```diff
@@ -0,0 +1,21 @@
+CHARACTERS_METADATA = {}
+
+
+def load_metadata():
```
可以参考一下我在隔壁对 `metadata` 元数据的装载，这样可以[避免全局变量的使用](https://wiki.c2.com/?GlobalVariablesAreBad)。也能保证使用的是同一个 `instance`
https://github.com/PaiGramTeam/python-genshin-artifact/blob/main/python_genshin_artifact/enka/assets.py

luoshuijs reviewed (commented):
剩下的就是CI指出的问题了， `PY-R1000` 可以不用管

luoshuijs committed:
:art: Use `items()` to iterate over a dictionary

luoshuijs committed:
:art: Use Unique Instance

luoshuijs committed:
:art: Optimize imports

luoshuijs committed:
:art: Optimize log

luoshuijs committed:
:art: `skipcq: PY-R1000`

luoshuijs committed:
:art: Fix `PYL-W0640`

luoshuijs committed:
:art: Fix `PYL-W0612`

omg-xtao committed:
fix: encoding

omg-xtao committed:
fix: bugs

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in dfe8a48 according to the output
from Black.

Details: https://github.com/PaiGramTeam/PaiGram/pull/709

luoshuijs committed:
:art: Update Code

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in plugins/genshin/gcsim/plugin.py with DeepSource Autofix

luoshuijs committed:
:art: Update Code

LeiShi1313 committed:
merge origin

omg-xtao committed:
fix: skill level

luoshuijs committed:
:art: remove `text` parameter

LeiShi1313 committed:
fix gcsim for <4 fits

LeiShi1313 committed:
Merge branch 'gcsim' of github.com:PaiGramTeam/PaiGram into gcsim

omg-xtao committed:
fix: ci

omg-xtao committed:
move generated files to data

omg-xtao committed:
feat: make scripts remote

NahidaBuer committed:
:bug: Fix reply to message

NahidaBuer committed:
:memo: Add command hint for `/gcsim`

NahidaBuer committed:
:bug: Fix reply to message

NahidaBuer committed:
:memo: Add command hint for `/set_command`

LeiShi1313 committed:
optimize aiofiles

omg-xtao committed:
fix: ci

LeiShi1313 committed:
fix enka parse error

omg-xtao committed:
clear fits when refresh scripts

LeiShi1313 committed:
add refresh button

LeiShi1313 committed:
fix ci

omg-xtao committed:
create cache

omg-xtao committed:
fix: ci

LeiShi1313 committed:
add trial tag

LeiShi1313 committed:
merge

LeiShi1313 committed:
fix ci

LeiShi1313 committed:
fix ci

omg-xtao committed:
add memory and time limit
